### PR TITLE
Fix bugs caused by legacy columns in the discussion model.

### PR DESCRIPTION
**Pull request by mtschirs**
- Removed legacy columns from the discussion model
- Replaced references to those legacy columns from discussion views and
the Tagging plugin as well as the IndexPhotos plugin

This fixes http://vanillaforums.org/discussion/30496/issue-recent-discussions-module-doesnt-show-author#latest

### CODE REVIEW

**Comment by linc** [comment]
Some serious heavy lifting in here, thanks. At first blush, I like this plan. I'm going to need to take a deeper dive on it and do some internal code vetting on our side to make sure this isn't going to mess up cloud, and I'm gonna want @tburry to take a look to make sure these query changes aren't gonna undo some sort of sidehatch performance gain I don't know about. This might take a bit.

**Comment by mtschirs** [comment]
Thank you for looking into it! Since I only removed superfluous joins from the existing queries, the performance *should* improve a bit. But who knows...

**Comment by linc** [comment]
This will absolutely clobber our VIP themes as it stands. Need to put a hold on this until we can get that resolved. Shouldn't be a huge deal to clean up, but it's definitely some time to invest and we'll need to wait for a release cycle after it's fixed everywhere.

File: applications/vanilla/models/class.discussionmodel.php [hunk 1]
```diff
@@ -654,14 +631,6 @@ public function calculate(&$Discussion) {
...
         $Discussion->PermissionCategoryID = $Category['PermissionCategoryID'];
 
-        // Add some legacy calculated columns.
-        if (!property_exists($Discussion, 'FirstUserID')) {
-            $Discussion->FirstUserID = $Discussion->InsertUserID;
-            $Discussion->FirstDate = $Discussion->DateInserted;
-            $Discussion->LastUserID = $Discussion->LastCommentUserID;
-            $Discussion->LastDate = $Discussion->DateLastComment;
-        }
-
```
Comment (tburry) [inline applications/vanilla/models/class.discussionmodel.php]: I think leaving something like this in for a period of time would help with the deprecation process.

File: applications/vanilla/models/class.discussionmodel.php [hunk 2]
```diff
@@ -716,11 +685,6 @@ public function calculate(&$Discussion) {
 
         $Discussion->CountCommentWatch = is_numeric($Discussion->CountCommentWatch) ? $Discussion->CountCommentWatch : null;
 
-        if ($Discussion->LastUserID == null) {
-            $Discussion->LastUserID = $Discussion->InsertUserID;
-            $Discussion->LastDate = $Discussion->DateInserted;
-        }
-
```
Comment (tburry) [inline applications/vanilla/models/class.discussionmodel.php]: You can see here that we set a user in the case of a discussion with no comments. Removing this kind of logic could lead to theme breakage.

**Comment by tburry** [comment]
I think the original intention was to remove that join, but we had forgot about it. I like these changes, but really want some measure of backwards compatibility for a transition period as I've noted in my line comments.

File: applications/vanilla/models/class.discussionmodel.php (hunk 1 above)
Comment (mtschirs) [inline applications/vanilla/models/class.discussionmodel.php]: I would assume those are already legacy (=deprecated) columns.

File: applications/vanilla/models/class.discussionmodel.php (hunk 2 above)
Comment (mtschirs) [inline applications/vanilla/models/class.discussionmodel.php]: This code is flawed and causes actual theme breakage right now (e.g. see vanillaforums.org/discussion/30496/issue-recent-discussions-module-doesnt-show-author).

That's what this PR is all about - remove these incomplete and actually already theme-breaking legacy calculated columns and adapt old themes to no longer rely on their presence.

Comment (tburry) [inline applications/vanilla/models/class.discussionmodel.php]: Ah OK. Fair enough.

**Pull request closed without merging**